### PR TITLE
[Functions] (refactor): Optimize struct packing

### PR DESCRIPTION
**Pull request by justinkaseman**
Gas golf: -1394 gas
(also a little more housecleaning + additional explainer comments)

### CODE REVIEW

**Review by KuphJr — DISMISSED** [review]

File: core/services/ocr2/plugins/functions/integration_tests/v1/functions_integration_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var (
 )
 
 func TestIntegration_Functions_MultipleV1Requests_Success(t *testing.T) {
-	t.Skip()
+	// t.Skip()
```
Comment (KuphJr) [inline core/services/ocr2/plugins/functions/integration_tests/v1/functions_integration_test.go:23]: ?

**Review by KuphJr — APPROVED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol [hunk 2]
```diff
@@ -63,9 +63,9 @@ contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable,
   error GasLimitTooBig(uint32 limit);
 
   struct CallbackResult {
-    bool success;
-    uint256 gasUsed;
-    bytes returnData;
+    bool success; // ══════╸ Whether the callback succeeded or not
```
Comment (RensR) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:66]: Nit: why the switch to double lines and lines for items that are not packed with other items? 

Double lines are more of a stylistic choice (and I don't care much about it) but I Personally find the lines for things that are unpacked fairly ugly. Won't be blocking the PR of course, all just decoration

**Review by RensR — APPROVED** [review]
Just stylistic differences ;)

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol (hunk 2 above)
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:66]: While maybe uglier, I find it a lot more clear. When there are no lines, at a glance it looks like the struct could have not been packed efficiently.

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:66]: Think of it as denoting the slot, rather than what is packed together

**Pull request merged**